nexus/db-queries/src/db/datastore/db_metadata.rs

```diff
@@ -4,23 +4,79 @@
 
 //! [`DataStore`] methods on Database Metadata.
 
-use super::DataStore;
+use super::{DataStore, DbConnection, IdentityCheckPolicy};
+use crate::authz;
+use crate::context::OpContext;
+
 use anyhow::{Context, bail, ensure};
 use async_bb8_diesel::{AsyncRunQueryDsl, AsyncSimpleConnection};
 use chrono::Utc;
 use diesel::prelude::*;
+use futures::FutureExt;
 use nexus_db_errors::ErrorHandler;
+use nexus_db_errors::OptionalError;
 use nexus_db_errors::public_error_from_diesel;
 use nexus_db_model::AllSchemaVersions;
+use nexus_db_model::DB_METADATA_NEXUS_SCHEMA_VERSION;
+use nexus_db_model::DbMetadataNexus;
+use nexus_db_model::DbMetadataNexusState;
 use nexus_db_model::EARLIEST_SUPPORTED_VERSION;
 use nexus_db_model::SchemaUpgradeStep;
 use nexus_db_model::SchemaVersion;
+use nexus_types::deployment::BlueprintZoneDisposition;
 use omicron_common::api::external::Error;
+use omicron_uuid_kinds::GenericUuid;
+use omicron_uuid_kinds::OmicronZoneUuid;
 use semver::Version;
 use slog::{Logger, error, info, o};
 use std::ops::Bound;
 use std::str::FromStr;
 
+/// Errors that can occur during handoff operations
```

Comment:
I expected that most of the changes in this file would be in #8925 and not this one.

Reply:
Moved into https://github.com/oxidecomputer/omicron/pull/8931